### PR TITLE
[IPP] Update stripe SDK

### LINKED ISSUE #8881: [Mobile Payments] Update the stripe SDK to the latest version (there are issues with the ROOM version)

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #8881

### Description

Updates Stripe SDK version. This brings updates of the many other dependencies, including Room and ~~Kotlin std lib~~ (updated separately)

@oguzkocer @wzieba do you have any concerns regarding this?

### Testing instructions

* Run TTP and IPP (both supported readers) flows with and without `stripe_repo_url`
* Simulate crash and verify that it's send to Sentry

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -27,7 +27,7 @@ pluginManagement {
         id 'com.google.gms.google-services' version '4.3.10'
         id 'io.github.wzieba.tracks.plugin' version '1.0.0'
         id 'io.gitlab.arturbosch.detekt' version gradle.ext.detektVersion
-        id 'io.sentry.android.gradle' version '3.1.3'
+        id 'io.sentry.android.gradle' version '3.5.0'
```
Comment (kidinov) [inline settings.gradle:30]: Older version has [this bug](https://github.com/getsentry/sentry-android-gradle-plugin/issues/434)

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuScreen.kt [hunk 2]
```diff
@@ -380,6 +380,7 @@ fun MoreMenuBadge(badgeState: BadgeState?) {
             verticalAlignment = Alignment.CenterVertically
         ) {
             Spacer(modifier = Modifier.weight(1f))
+            @Suppress("RememberReturnType")
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/moremenu/MoreMenuScreen.kt]: False positive caused by [the bug](https://issuetracker.google.com/issues/263887242)

**Comment by oguzkocer** [comment]
> @oguzkocer @wzieba do you have any concerns regarding this?

I unfortunately don't have enough context to suggest or discourage this update. One thing we try to do with dependency updates is to try to avoid too many transitive updates and instead update them ourselves. The problem with that approach is that it takes time to go through all the updates, so it's not a feasible choice when there is a time constraint. With that in mind, I think Woo Android team is in a better position to decide whether it's better to merge this PR as is, or try to update the transitive dependencies first.

I'll also cc @ParaskP7 who has been handling most of the library updates in our team, so he is the best person to check with.

**Comment by ParaskP7** [comment]
👋 @kidinov and thanks for the ping @oguzkocer ! 👍

Oguz is right, I would personally recommend updating `Kotlin` + `Compose Compiler` first, test that update in isolation, which will require smoke testing the whole app, and only then update the `Stripe SDK` and focus test this update.

Also, since this PR updates `Sentry` too, this should be tested as well.

In WPAndroid, I make sure to provide specific testing instructions per update in order to make sure that nothing vital is missed, causing crashes or misbehavior further down the line.

Having said that, I know that the team doesn't always have the luxury to follow this process, due to time constraints, business needs and what not. Thus, I am letting the team decide if they are comfortable merging this to `trunk`, given they have tested all the updates, transient ones included. 🤷

**Comment by kidinov** [comment]
@ParaskP7 @oguzkocer Thanks!

> Oguz is right, I would personally recommend updating Kotlin + Compose Compiler first,

How do you see that Compose Compiler was updated?

**Comment by ParaskP7** [comment]
👋 @kidinov !

> How do you see that Compose Compiler was updated?

No, I didn't see the `Compose Compiler` get updated, I just mention that because updating `Kotlin` would require you to update `Compose Compiler` as well, since those two libraries are bound together... 🤷

**Review by samiuelson — APPROVED** [review]
I smoke-tested the payment flows and critical flows. LGTM :shipit:

**Pull request merged**